image-rauc.c

```diff
@@ -36,7 +37,9 @@ static int rauc_generate(struct image *image)
 	const char *cert = cfg_getstr(image->imagesec, "cert");
 	const char *key = cfg_getstr(image->imagesec, "key");
 	const char *keyring = cfg_getstr(image->imagesec, "keyring");
-	char *keyringarg = "";
+	const char *intermediate = cfg_getstr(image->imagesec, "intermediate");
```

Comment:
So we have 4 options here that are treated similar but not quite the same way:

- `cert` and `key` can be either a file (handled as a partition) or a pkcs11uri
- `keyring` and now `intermediate` must be a file.

Is this correct? @jluebbe can you comment on this?
If it must be a file, then we don't need to read the config value here, because there will be a partition below that overwrites it. This is already wrong for `keyring`.

Reply:
`--intermediate` also accepts a pkcs11-URL. Note that you can specify it multiple times (if you have a chain of CA, intermediate 1, intermediate 2, leaf).

Reply:
Right, so that should be a string list instead. Is using `--intermediate` multiple times relevant for files as well, or only for pkcs11? I'd expect the whole chain in one file.

Hmm, what about keyring? can this be a pkcs11-URL as well?

Reply:
I am not sure what that will look like in config file.

```
image @IMAGE@ {
   rauc {
      intermediate = "cert1.pem cert2.pem"
   }
}
```

something like that?

Reply:
`intermediate = { "cert1.pem", "cert2.pem" }'

Take a look at the vfat `files` option. That uses a string list.

Hmmm, I think the code need to change: Maybe loop over the list and add arguments for all pkcs11-URLs. Then loop over the partitions and add the arguments for the files or something like that.

Reply:
`--keyring` must be a file (as it can contain multiple certificates). Each `--intermediate` must only refer to a single certificate (in both the file and pkcs11 cases).

The keyring is used only to verify the resulting signature.